src/test/java/de/tum/in/www1/artemis/service/compass/controller/AutomaticAssessmentControllerTest.java

```diff
@@ -52,134 +53,218 @@
         when(classDiagram.getElementByJSONID("element2Id")).thenReturn(null);
 
-        automaticAssessmentController.addFeedbackToSimilaritySet(assessmentIndex, elementIdFeedbackMap, classDiagram);
+        automaticAssessmentController.addFeedbacksToSimilaritySet(feedbacks, classDiagram);
+        Optional<SimilaritySetAssessment> ssA = automaticAssessmentController.getAssessmentForSimilaritySet(1);
 
-        verify(similaritySetAssessment, never()).addFeedback(any(Feedback.class));
-        verify(assessmentIndex, never()).addSimilaritySetAssessment(anyInt(), any(SimilaritySetAssessment.class));
+        assertThat(automaticAssessmentController.getAssessmentMap().values().size()).isEqualTo(0);
     }
 
     @Test
     void assessModelsAutomatically() {
         automaticAssessmentController = mock(AutomaticAssessmentController.class);
-        doCallRealMethod().when(automaticAssessmentController).assessModelsAutomatically(modelIndex, assessmentIndex);
-        when(automaticAssessmentController.assessModelAutomatically(classDiagram, assessmentIndex)).thenReturn(mock(CompassResult.class));
-        when(automaticAssessmentController.assessModelAutomatically(activityDiagram, assessmentIndex)).thenReturn(mock(CompassResult.class));
+        doCallRealMethod().when(automaticAssessmentController).assessModelsAutomatically(modelIndex);
+        when(automaticAssessmentController.assessModelAutomatically(classDiagram)).thenReturn(mock(CompassResult.class));
+        when(automaticAssessmentController.assessModelAutomatically(activityDiagram)).thenReturn(mock(CompassResult.class));
         when(modelIndex.getModelCollection()).thenReturn(List.of(classDiagram));
         when(modelIndex.getModelCollection()).thenReturn(List.of(classDiagram, activityDiagram));
 
-        automaticAssessmentController.assessModelsAutomatically(modelIndex, assessmentIndex);
+        automaticAssessmentController.assessModelsAutomatically(modelIndex);
 
-        verify(automaticAssessmentController).assessModelAutomatically(classDiagram, assessmentIndex);
-        verify(automaticAssessmentController).assessModelAutomatically(activityDiagram, assessmentIndex);
+        verify(automaticAssessmentController).assessModelAutomatically(classDiagram);
+        verify(automaticAssessmentController).assessModelAutomatically(activityDiagram);
     }
 
     @Test
-    void assessModelAutomatically_ClassDiagram() {
+    void assessModelAutomaticallyClassDiagram() {
         prepareClassDiagramForAutomaticAssessment();
         prepareAssessmentIndexForAutomaticAssessment();
 
-        CompassResult compassResult = automaticAssessmentController.assessModelAutomatically(classDiagram, assessmentIndex);
+        CompassResult compassResult = automaticAssessmentController.assessModelAutomatically(classDiagram);
 
         assertThat(compassResult.entitiesCovered()).isEqualTo(6);
         assertThat(compassResult.getPoints()).isEqualTo(-0.5 - 0.5 + 0 + 1.5 + 1.0 + 0.5);
```

Comment:
I detect that this code is problematic. According to the [Correctness (CORRECTNESS)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#correctness-correctness), [SA: Nonsensical self computation involving a variable (e.g., x & x) (SA_LOCAL_SELF_COMPUTATION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#sa-nonsensical-self-computation-involving-a-variable-e-g-x-x-sa-local-self-computation).
This method performs a nonsensical computation of a local variable with another reference to the same variable (e.g., x&x or x-x). Because of the nature of the computation, this operation doesn't seem to make sense, and may indicate a typo or a logic error. Double check the computation.